### PR TITLE
chore: braze proxy v1 test

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?

Refactor proxy v1 tests for Braze Integration

## What is the related Linear task?

Resolves INT-1478

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3087/test-report.html

File: test/integrations/destinations/braze/dataDelivery/business.ts [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+              },
+              {
+                error:
+                  '{"events_processed":1,"message":"success","errors":[{"type":"\'external_id\', \'braze_id\', \'user_alias\', \'email\' or \'phone\' is required","input_array":"events","index":1},{"type":"\'quantity\' is not valid","input_array":"purchases","index":0}]}',
+                statusCode: 200,
+                metadata: generateMetadata(2),
+              },
+              {
+                error:
+                  '{"events_processed":1,"message":"success","errors":[{"type":"\'external_id\', \'braze_id\', \'user_alias\', \'email\' or \'phone\' is required","input_array":"events","index":1},{"type":"\'quantity\' is not valid","input_array":"purchases","index":0}]}',
```
Comment (koladilip) [inline test/integrations/destinations/braze/dataDelivery/business.ts:234]: repeated

File: test/integrations/destinations/braze/dataDelivery/business.ts [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    },
+    output: {
+      response: {
+        status: 200,
+        body: {
+          output: {
+            response: [
+              {
+                error:
+                  '{"message":"Valid data must be provided in the \'attributes\', \'events\', or \'purchases\' fields.","errors":[{"type":"\'external_id\', \'braze_id\', \'user_alias\', \'email\' or \'phone\' is required","input_array":"events","index":0},{"type":"\'external_id\', \'braze_id\', \'user_alias\', \'email\' or \'phone\' is required","input_array":"events","index":1},{"type":"\'quantity\' is not valid","input_array":"purchases","index":0}]}',
```
Comment (koladilip) [inline test/integrations/destinations/braze/dataDelivery/business.ts:283]: repeated

File: test/integrations/destinations/braze/dataDelivery/other.ts [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                  workspaceId: 'default-workspaceId',
+                  sourceId: 'default-sourceId',
+                  secret: {
+                    accessToken: 'default-accessToken',
+                  },
+                  dontBatch: false,
+                },
+              },
+            ],
+            statTags: {
```
Comment (koladilip) [inline test/integrations/destinations/braze/dataDelivery/other.ts:47]: can we create a reusable function for this

File: test/integrations/destinations/braze/dataDelivery/other.ts [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        status: 200,
+        body: {
+          output: {
+            response: [
+              {
+                error:
+                  '{"error":{"message":"Service Unavailable","description":"The server is currently unable to handle the request due to temporary overloading or maintenance of the server. Please try again later."}}',
+                statusCode: 503,
+                metadata: {
+                  jobId: 1,
```
Comment (koladilip) [inline test/integrations/destinations/braze/dataDelivery/other.ts:34]: can we use getMetadata(1) here?

File: test/integrations/destinations/braze/dataDelivery/other.ts [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+      response: {
+        status: 200,
+        body: {
+          output: {
+            response: [
+              {
+                error: '"Internal Server Error"',
+                statusCode: 500,
+                metadata: {
+                  jobId: 1,
```
Comment (koladilip) [inline test/integrations/destinations/braze/dataDelivery/other.ts:91]: getMetadata(1)

File: test/integrations/destinations/braze/network.ts [hunk 6]
```diff
@@ -524,4 +524,225 @@ const deleteNwData = [
...
+  name: 'Product List Viewed',
+  time: '2023-11-30T21:48:45.634Z',
+  properties: {
+    products: [
+      {
+        sku: '23-04-52-62-01-18',
+        name: 'Broman Hoodie',
+        price: '97.99',
+        variant: [
+          {
```
Comment (koladilip) [inline test/integrations/destinations/braze/network.ts:542]: looks repeated

File: test/integrations/destinations/braze/dataDelivery/business.ts (hunk 1 above)
Comment (utsabc) [reply test/integrations/destinations/braze/dataDelivery/business.ts]: addressed

File: test/integrations/destinations/braze/dataDelivery/business.ts (hunk 2 above)
Comment (utsabc) [reply test/integrations/destinations/braze/dataDelivery/business.ts]: addressed

File: test/integrations/destinations/braze/dataDelivery/other.ts (hunk 3 above)
Comment (utsabc) [reply test/integrations/destinations/braze/dataDelivery/other.ts]: declared a common variable

File: test/integrations/destinations/braze/dataDelivery/other.ts (hunk 4 above)
Comment (utsabc) [reply test/integrations/destinations/braze/dataDelivery/other.ts]: addressed

File: test/integrations/destinations/braze/dataDelivery/other.ts (hunk 5 above)
Comment (utsabc) [reply test/integrations/destinations/braze/dataDelivery/other.ts]: addressed

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3087/test-report.html

File: test/integrations/destinations/braze/network.ts (hunk 6 above)
Comment (utsabc) [reply test/integrations/destinations/braze/network.ts]: these were not used hence cleaned up

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3087/test-report.html

**Pull request merged**